swupd/create_manifests.go

```diff
@@ -52,12 +52,15 @@ func getOldManifest(path string) (*Manifest, error) {
 	return ParseManifestFile(path)
 }
 
-func initBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
+func initBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error) {
 	var wg sync.WaitGroup
-	workers := len(ui.bundles)
-	wg.Add(workers)
+	if numWorkers > len(ui.bundles) || numWorkers == 0 {
```

Comment:
`numWorkers` is a signed `int` so you need to check if it is `<= 0` not just `== 0`.

Reply:
although if it is `<= 0` you might want to set to `1` instead of `len(ui.bundles)`

Reply:
hmm, maybe I should just change the type instead. It makes no sense having numWorkers < 0

Reply:
sure, that would make sense, but you'll have to change it in the entire call path from the mixer cmdline as well if you do that.

Reply:
No problem. It looks better overall.

Reply:
ok +1

Reply:
hmm, nevermind... doing so would require me to manually convert the result of runtime.NumCPU() to uint in multiple points of the code. It will be less clunky to just handle <= 0 here.